config.go

```diff
@@ -143,11 +145,26 @@ func getEffectiveConfig(configRoot, owner, repo string) (*effectiveConfig, error
 }
 
 func buildGlobalEffectiveConfig(global *config) effectiveConfig {
+	origServerID := firstNonEmptyStringPtr(global.GrimServerID, global.GrimQueueName, &defaultGrimQueueName)
+	truncatedServerID := origServerID
+	truncateID := ""
+
+	if len(truncatedServerID) > 15 {
+		truncatedServerID = fmt.Sprintf("%s", origServerID[:15])
+
+		truncateID = "GrimServerID"
+		if !stringPtrNotEmpty(global.GrimServerID) {
+			truncateID = "GrimQueueName"
+		}
+	}
+
```

Comment:
Personally I think I'd like to see this pulled out into a function called `performCorrections` (or something to that effect) that takes an `effectiveConfig` and returns `(effectiveConfig, []error)`.  That way the caller has the before/after and the messages that it can log or throw away.

Don't worry about it unless @kklipsch agrees though.  This implementation is solid as is.